pkg/block/fetcher.go

```diff
@@ -283,9 +291,22 @@ func (f *ConcurrentLister) GetActiveAndPartialBlockIDs(ctx context.Context, ch c
 	}
 	close(metaChan)
 
-	if err := eg.Wait(); err != nil {
-		return nil, err
+	multiError = multierror.Append(eg.Wait(), multiError)
+
+	if multiError != nil {
+		if multiErrorUnwrap, ok := err.(interface{ Unwrap() []error }); ok {
```

Comment:
This should always succeed unless we change the library, right? Perhaps in the `!ok` clause we could put a `panic("BUG: multierr doesn't implement unwrap")` to make sure that in the future we would notice this part if suddenly `err` wouldn't implement this interface?

Reply:
Added the panic with a comment. Seeing that Unwrap is basically the core feature that go-multierror adds to the mix this should never happen. The panic therefore would be absolutely warranted if it still happens. :)

Reply:
I ran into a bunch of issues mostly due to the panic. I then realised that this could be so much simpler, and I refactored it so that the panic is no longer necessary, it would just give a compile error if something major changed in the imported package.

I also realised that my test case's success condition was incorrect. I cannot assert on no errors occuring, since the whole point is that simulated errors are occurring. What I should (and now I am) testing for is that I get exactly the expected number of errors. If there was a deadlock, it would be missing some of them.

I ended up squashing the entire change-set as I forgot to sign the commits. ( :/ )